Sources/TuistCore/Utils/XCFrameworkMetadataProvider.swift

```diff
@@ -0,0 +1,26 @@
+import Basic
+import Foundation
+import TuistSupport
+
+public protocol XCFrameworkMetadataProviding {
+    func libraries(
+        frameworkPath: AbsolutePath,
+        fileHandler: FileHandling
+    ) throws -> [XCFrameworkInfoPlist.Library]
+}
+
+public class XCFrameworkMetadataProvider: XCFrameworkMetadataProviding {
+
+    public func libraries(
+        frameworkPath: AbsolutePath,
+        fileHandler: FileHandling = FileHandler.shared
+    ) throws -> [XCFrameworkInfoPlist.Library] {
+        let infoPlist = frameworkPath.appending(component: "Info.plist")
+        guard fileHandler.exists(infoPlist) else {
+            throw GraphLoadingError.missingFile(infoPlist)
```

Comment:
If the code execution reaches this point and we tell the user:

> File not found at /whatever/whatever/framework.xcframework/Info.plist

I don't think most users would know what to do. I suggest defining a `XCFrameworkMetadataProviderError` and use a more specific error & message:


> The .xcframework at path /whatever/whatever/framework.xcframework doesn't contain an Info.plist. It's possible that the .xcframework was not generated properly or that got corrupted. Please, double check with the author of the framework.